### PR TITLE
Add Mac compatibility to integration tests

### PR DESCRIPTION
**Pull request by BKreisel**
Adds OS X compatibility to integration tests by addressing #680, #857, and #724.

greadlink dependency introduced for OS X will be installed with #883.

### CODE REVIEW

File: letsencrypt/plugins/manual.py [hunk 1]
```diff
@@ -159,6 +159,7 @@ def _perform_single(self, achall):
                     # don't care about setting stdout and stderr,
                     # we're in test mode anyway
                     shell=True,
+                    executable="/bin/bash",
```
Comment (kuba) [inline letsencrypt/plugins/manual.py]: Careful, some systems don't have `bash` out of the box, e.g. FreeBSD. Could we make `executable="/bin/bash"` conditional on Mac OS X presence? Shouldn't it be installed in #883 bootstrap scripts?

File: letsencrypt/plugins/standalone/authenticator.py [hunk 2]
```diff
@@ -301,11 +301,14 @@ def already_listening(self, port):  # pylint: disable=no-self-use
         :param int port: The TCP port in question.
         :returns: True or False."""
 
-        listeners = [conn.pid for conn in psutil.net_connections()
-                     if conn.status == 'LISTEN' and
-                     conn.type == socket.SOCK_STREAM and
-                     conn.laddr[1] == port]
         try:
```
Comment (kuba) [inline letsencrypt/plugins/standalone/authenticator.py]: [PEP8](https://www.python.org/dev/peps/pep-0008/): "Additionally, for all try/except clauses, limit the try clause to the absolute minimum amount of code necessary. Again, this avoids masking bugs."

This clause has grown too much now. Please refactor something like:

```python
try:
    net_connections = psutil.net_connections()
except psutil.AccessDenied as error:
    logger.info("Access denied when trying to list network connections: "
                "%s. Are you root?", error)
    # this function is just a pre-check that often causes false positives
    # and problems in testing (c.f. #680 on Mac, #255 generally); we will
    # fail later in bind() anyway
    return False

try:
    listeners = [... for conn in net_connections ...]
...
```

In fact everything else apart from `psutil.Process(pid).name()` should be taken out of the `try/except` block, but this is out of scope for this PR.

File: tests/boulder-integration.sh [hunk 3]
```diff
@@ -50,7 +50,11 @@ dir="$root/conf/archive/le1.wtf"
 for x in cert chain fullchain privkey;
 do
     latest="$(ls -1t $dir/ | grep -e "^${x}" | head -n1)"
-    live="$(readlink -f "$root/conf/live/le1.wtf/${x}.pem")"
+    if [ `uname`  == 'Darwin' ]; then
+      live="$(greadlink -f "$root/conf/live/le1.wtf/${x}.pem")"
+    else
+      live="$(readlink -f "$root/conf/live/le1.wtf/${x}.pem")"
+    fi
```
Comment (kuba) [inline tests/boulder-integration.sh]: I often find that defining at the top of the file
```bash
if [ `uname`  == 'Darwin' ]; then
  readlink="greadlink"
else
  readlink="readlink"
fi
```

and using it later as `$readlink -f ...` nicely deduplicates, leaving less space for error and maintenance burden.

**Comment by kuba** [comment]
LGTM modulo fixes above (especially `bash` stuff).

**Comment by BKreisel** [comment]
I've been PEP8'ed!  Thanks for the advice @kuba. Your comments had great advice.

File: letsencrypt/plugins/manual.py [hunk 4]
```diff
@@ -153,12 +153,19 @@ def _perform_single(self, achall):
             ct=response.CONTENT_TYPE, port=port)
         if self.conf("test-mode"):
             logger.debug("Test mode. Executing the manual command: %s", command)
+            # sh shipped with OS X does't support echo -n
+            if sys.platform == "darwin":
+                executable = "/bin/bash"
+            else:
+                executable = "/bin/sh"
```
Comment (kuba) [inline letsencrypt/plugins/manual.py]: I think this should be `executable = None` to make sure platform's default is used. https://docs.python.org/2/library/subprocess.html#popen-constructor

File: letsencrypt/plugins/standalone/authenticator.py [hunk 5]
```diff
@@ -301,10 +305,21 @@ def already_listening(self, port):  # pylint: disable=no-self-use
         :param int port: The TCP port in question.
         :returns: True or False."""
 
-        listeners = [conn.pid for conn in psutil.net_connections()
+        try:
+            net_connections = psutil.net_connections()
+        except psutil.AccessDenied as error:
+            logger.info("Access denied when trying to list network "
+            "connections: %s. Are you root?", error)
```
Comment (kuba) [inline letsencrypt/plugins/standalone/authenticator.py]: This is missing indent (should match opening quote from line above), but I don't know why pylint didn't catch it...

**Comment by kuba** [comment]
LGTM, thanks a lot! :)

Two small nits can be fixed by the merger.

**Comment by bmw** [comment]
@jmhodges does this fix integration testing for you?

**Comment by kuba** [comment]
@BKreisel: lint unhappy: https://travis-ci.org/letsencrypt/letsencrypt/jobs/84379558#L1030:
```
************* Module letsencrypt.plugins.manual
C:137, 0: Trailing whitespace (trailing-whitespace)
```

**Comment by bmw** [comment]
LGTM! Thanks for another PR @BKreisel. We certainly appreciate it.

**Pull request merged**